File: src/main/java/org/example/homework/service/ReportService.java
package org.example.homework.service;

import org.example.homework.dto.response.AvailableResourceItem;
import org.example.homework.dto.response.OverloadedEmployeeItem;
import org.example.homework.dto.response.UtilizationReportItem;
import java.util.List;

public interface ReportService {
    List<UtilizationReportItem> getUtilizationReport();
    List<AvailableResourceItem> getAvailableResourcesReport();
    List<OverloadedEmployeeItem> getOverloadedEmployeesReport();
}

File: frontend/src/app/services/report.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { UtilizationReportItem, AvailableResourceItem, OverloadedEmployeeItem } from '../models';

@Injectable({
  providedIn: 'root'
})
export class ReportService {
  private baseUrl = 'http://localhost:8080/api/reports';

  constructor(private http: HttpClient) {}

  getUtilizationReport(): Observable<UtilizationReportItem[]> {
    return this.http.get<UtilizationReportItem[]>(`${this.baseUrl}/utilization`);
  }

  getAvailableResourcesReport(): Observable<AvailableResourceItem[]> {
    return this.http.get<AvailableResourceItem[]>(`${this.baseUrl}/available-resources`);
  }

  getOverloadedEmployeesReport(): Observable<OverloadedEmployeeItem[]> {
    return this.http.get<OverloadedEmployeeItem[]>(`${this.baseUrl}/overloaded`);
  }
}
